tests/chainer_tests/links_tests/connection_tests/test_convolution_2d.py

```diff
@@ -305,17 +300,59 @@ def test_forward(self, backend_config):
             (self.height - self.kernel_height + 1) // self.strides_height,
             (self.width - self.kernel_width + 1) // self.strides_width)
 
-    def test_forward_invalid_backend(self):
+
+@testing.inject_backend_tests(
+    [
+        'test_forward',
+    ],
+    # CPU tests
+    [{}]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'use_cudnn': ['never'],
+    }))
+class TestConvolution2DMemoryLayoutsInvalidBackend(unittest.TestCase):
+    batch = 2
+    in_channels = 5
+    out_channels = 7
+    height = 13
+    width = 11
+    kernel_height = 5
+    kernel_width = 4
+    strides_height = 3
+    strides_width = 2
+    dtype = numpy.float32
```

Comment:
Rather than repeating test parameters, how about using the single class and change the behavior within the test?
Or, how about creating the base class with these parameters, and deriving each class from it?